### PR TITLE
Update for 2.4.0

### PR DESCRIPTION
**Pull request by adam-urbanczyk**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by adam-urbanczyk** [comment]
@conda-forge-admin, please rerender

**Comment by adam-urbanczyk** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -15,20 +16,20 @@ build:
...
   host:
-    - python >=3.8
+    - python >=3.9
     - pip
   run:
-    - casadi
+    - python >=3.9
+    - ocp 7.7.2
+    - vtk =*=qt*
+    - pyparsing >=2.1.9
```
Comment (bollwyvl) [inline recipe/meta.yaml:24]: please consider keeping this list sorted so it's easier to see what changed over time

File: recipe/meta.yaml [hunk 2]
```diff
@@ -15,20 +16,20 @@ build:
 
 requirements:
   host:
-    - python >=3.8
+    - python >=3.9
```
Comment (bollwyvl) [inline recipe/meta.yaml:19]: upstream still says 3.8... https://github.com/CadQuery/cadquery/blob/master/setup.py#L51

File: recipe/meta.yaml [hunk 3]
```diff
@@ -15,20 +16,20 @@ build:
...
     - pip
   run:
-    - casadi
+    - python >=3.9
+    - ocp 7.7.2
+    - vtk =*=qt*
```
Comment (bollwyvl) [inline recipe/meta.yaml]: is it possible to run without vtk/qt? if so, this should likely be in a separate package in the `outputs` section

**Comment by adam-urbanczyk** [comment]
@bollwyvl are you OK with merging?

File: recipe/meta.yaml (hunk 2 above)
Comment (bollwyvl) [reply recipe/meta.yaml:19]: again, what is the motivation for not using he upstream pin?

File: recipe/meta.yaml (hunk 1 above)
Comment (bollwyvl) [reply recipe/meta.yaml:24]: again, sorting the list of dependencies would make this easier to reason about.

**Comment by adam-urbanczyk** [comment]
So are w good here?

**Comment by adam-urbanczyk** [comment]
Green build and no reply to my clarifications - going ahead with merging.

**Pull request merged**